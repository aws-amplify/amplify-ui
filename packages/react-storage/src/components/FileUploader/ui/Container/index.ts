export { Container, ContainerProps } from './Container';
