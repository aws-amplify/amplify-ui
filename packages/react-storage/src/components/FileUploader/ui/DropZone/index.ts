export { DropZone } from './DropZone';
export { DropZoneProps } from './types';
