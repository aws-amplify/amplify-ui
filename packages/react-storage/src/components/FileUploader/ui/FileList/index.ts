export { FileList } from './FileList';
export { FileListProps } from './types';
