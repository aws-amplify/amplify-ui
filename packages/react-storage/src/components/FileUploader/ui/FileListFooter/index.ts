export type { FileListFooterProps } from './FileListFooter';
export { FileListFooter } from './FileListFooter';
