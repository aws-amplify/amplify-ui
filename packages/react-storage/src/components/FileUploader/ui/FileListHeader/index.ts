export { FileListHeader, FileListHeaderProps } from './FileListHeader';
