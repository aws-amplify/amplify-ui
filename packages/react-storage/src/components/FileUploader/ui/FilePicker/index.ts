export { FilePicker, FilePickerProps } from './FilePicker';
