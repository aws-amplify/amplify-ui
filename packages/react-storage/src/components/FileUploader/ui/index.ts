export { Container, ContainerProps } from './Container';
export { DropZone, DropZoneProps } from './DropZone';
export { FileList, FileListProps } from './FileList';
export { FileListHeader, FileListHeaderProps } from './FileListHeader';
export { FileListFooter, FileListFooterProps } from './FileListFooter';
export { FilePicker, FilePickerProps } from './FilePicker';
