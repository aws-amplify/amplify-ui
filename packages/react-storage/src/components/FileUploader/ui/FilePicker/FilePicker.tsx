import React from 'react';

import { ComponentClassName } from '@aws-amplify/ui';
import { Button, ButtonProps } from '@aws-amplify/ui-react';

export type FilePickerProps = ButtonProps;

export function FilePicker({
  children,
  className = ComponentClassName.FileUploaderFilePicker,
  size = 'small',
  ...props
}: FilePickerProps): JSX.Element {
  return (
    <Button {...props} className={className} size={size}>
      {children}
    </Button>
  );
}
