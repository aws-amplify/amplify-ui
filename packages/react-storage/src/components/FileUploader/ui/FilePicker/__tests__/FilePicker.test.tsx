import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { ComponentClassName } from '@aws-amplify/ui';

import { FilePicker } from '../FilePicker';

const children = 'Pick a file, any file.';
const onClick = jest.fn();

describe('FilePicker', () => {
  it('renders correctly', () => {
    const { container } = render(
      <FilePicker onClick={onClick}>{children}</FilePicker>
    );

    expect(container).toMatchSnapshot();
  });

  it('shows correct classname', async () => {
    render(<FilePicker onClick={onClick}>{children}</FilePicker>);

    const filePickerButton = await screen.findByRole('button');
    expect(filePickerButton).toHaveClass(
      ComponentClassName.FileUploaderFilePicker
    );
  });
});
