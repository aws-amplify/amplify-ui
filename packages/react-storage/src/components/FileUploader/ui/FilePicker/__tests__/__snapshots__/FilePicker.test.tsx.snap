// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePicker renders correctly 1`] = `
<div>
  <button
    class="amplify-button amplify-field-group__control amplify-button--small amplify-fileuploader__file__picker"
    type="button"
  >
    Pick a file, any file.
  </button>
</div>
`;
